File: rebus-rebuscrus/src/java/it/csi/rebus/rebuscrus/vo/AllegatoProcVO.java
/*
* SPDX-FileCopyrightText: (C) Copyright 2022 Regione Piemonte
* SPDX-License-Identifier: EUPL-1.2
*/
package it.csi.rebus.rebuscrus.vo;

import java.util.Date;

@SuppressWarnings("serial")
public class AllegatoProcVO extends ParentVO {
	private Long idProcedimento;
	private TipoDocumentoVO tipoDocumento;
	private byte[] file;
	private String nomeFile;
	private Date dataCaricamento;

	public Long getIdProcedimento() {
		return idProcedimento;
	}

	public void setIdProcedimento(Long idProcedimento) {
		this.idProcedimento = idProcedimento;
	}

	public TipoDocumentoVO getTipoDocumento() {
		return tipoDocumento;
	}

	public void setTipoDocumento(TipoDocumentoVO tipoDocumento) {
		this.tipoDocumento = tipoDocumento;
	}

	public byte[] getFile() {
		return file;
	}

	public void setFile(byte[] file) {
		this.file = file;
	}

	public String getNomeFile() {
		return nomeFile;
	}

	public void setNomeFile(String nomeFile) {
		this.nomeFile = nomeFile;
	}

	public Date getDataCaricamento() {
		return dataCaricamento;
	}

	public void setDataCaricamento(Date dataCaricamento) {
		this.dataCaricamento = dataCaricamento;
	}

	public String getNote() {
		return note;
	}

	public void setNote(String note) {
		this.note = note;
	}

	private String note;
}

File: rebus-rebuscruswcl/src/app/vo/allegato-proc-vo.ts
/*************************************************
Copyright Regione Piemonte - 2022
SPDX-License-Identifier: EUPL-1.2-or-later
***************************************************/
import { TipoDocumentoVO } from "./extend-vo";

export class AllegatoProcVO {
    idProcedimento: number;
    tipoDocumento: TipoDocumentoVO;
    file: File;
    nomeFile: string;
    dataCaricamento: Date;
    note: string;
    isFileUploaded: boolean;

    constructor() {
        if (!this.tipoDocumento) this.tipoDocumento = new TipoDocumentoVO();
    }
}